### PR TITLE
ENH: expose snap

### PR DESCRIPTION
**Pull request by martinfleis**
xref #2010

I have also noticed that the changelog is not properly linted and did that while adding a note for the PR.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,27 +9,31 @@ Notes on dependencies:
   rtree package has also been removed. (#3035)
 
 New methods:
-- Added `count_coordinates` method from shapely to GeoSeries/GeoDataframe (#3026).
-- Added `minimum_clearance` method from shapely to GeoSeries/GeoDataframe (#2989).
-- Added `is_ccw` method from shapely to GeoSeries/GeoDataframe (#3027).
+
+- Added ``count_coordinates`` method from shapely to GeoSeries/GeoDataframe (#3026).
```
Comment (m-richards) [inline CHANGELOG.md:13]: None of these quote changes should be needed after #2662, they all seem to be rendered fine as is for me?

File: geopandas/base.py [hunk 2]
```diff
@@ -3555,6 +3555,109 @@ def shortest_line(self, other, align=True):
...
+        Where possible, this operation tries to avoid creating invalid geometries;
+        however, it does not guarantee that output geometries will be valid. It is the
+        responsibility of the caller to check for and handle invalid geometries.
+
+        Because too much snapping can result in invalid geometries being created,
+        heuristics are used to determine the number and location of snapped vertices
+        that are likely safe to snap. These heuristics may omit some potential snaps
+        that are otherwise within the tolerance.
+
+        The operation works on a 1-to-1 row-wise manner:
```
Comment (m-richards) [inline geopandas/base.py:3576]:
```suggestion
        The operation works in a 1-to-1 row-wise manner:
```

File: geopandas/base.py [hunk 3]
```diff
@@ -3555,6 +3555,109 @@ def shortest_line(self, other, align=True):
...
+
+        .. image:: ../../../_static/binary_op-01.svg
+        :align: center
+
+        Parameters
+        ----------
+        other : Geoseries or geometric object
```
Comment (m-richards) [inline geopandas/base.py:3583]:
```suggestion
        other : GeoSeries or geometric object
```

File: CHANGELOG.md (hunk 1 above)
Comment (martinfleis) [reply CHANGELOG.md:13]: Ah, true! I forgot we no longer embed the markdown in RST. I will revert those.

Comment (martinfleis) [reply CHANGELOG.md]: I've kept the empty lines. Those are not strictly required to properly render but are recommended by markdownlint.

**Review by m-richards — APPROVED** [review]

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Pull request merged**